### ISSUE TITLE
Pyright incorrectly rejects overriden methods in named tuple subclasses

### ISSUE THREAD

**Issue by fergusq** [microsoft/pyright#7256]
**Describe the bug**

It seems that the implementation of #7198 which disallows overriding named tuple fields also disallows overriding methods. This is a bug, as Python allows overriding methods of named tuple classes without issues.

**Code or Screenshots**

```py
from typing import NamedTuple

class a(NamedTuple):
    x = 0
    def f(self):
        print(self.x)

class b(a):
    def f(self):
        print(self.x + 1)

a().f()  # prints 0
b().f()  # prints 1
```

The above code works, yet pyright gives the following error:

```
/tmp/test.py
  /tmp/test.py:9:5 - error: Cannot override "f" because parent class "a" is a named tuple (reportIncompatibleVariableOverride)
1 error, 0 warnings, 0 informations
```

**Version**

Command line, version 1.1.350.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
